src/NServiceBus.AcceptanceTests/Sagas/When_simple_saga.cs

```diff
@@ -0,0 +1,101 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_simple_saga : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_execute_saga()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<SagaEndpoint>(b => b
+                    .When(session => session.SendLocal(new StartSagaMessage
+                    {
+                        Key = "Part1_Part2"
+                    }))
+                    .When(c => c.FirstMessageReceived, session => session.SendLocal(new OtherMessage
+                    {
+                        Part1 = "Part1",
+                        Part2 = "Part2"
+                    })))
+                .Done(c => c.SecondMessageReceived)
+                .Run();
+
+            Assert.IsTrue(context.SecondMessageReceived);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FirstMessageReceived { get; set; }
+            public bool SecondMessageReceived { get; set; }
+            public Guid SagaId { get; set; }
+        }
+
+        public class SagaEndpoint : EndpointConfigurationBuilder
+        {
+            public SagaEndpoint()
+            {
+                //note: the concurrency checks for the InMemory persister doesn't seem to work so limiting to 1 for now
+                EndpointSetup<DefaultServer>(c => c.LimitMessageProcessingConcurrencyTo(1));
+            }
+
+            public class Saga : SimpleSaga<Saga.SagaData>,
+                IAmStartedByMessages<StartSagaMessage>,
+                IAmStartedByMessages<OtherMessage>
+            {
+
+                protected override Expression<Func<SagaData, object>> CorrelationProperty
+                {
+                    get { return data => data.CorrelationId; }
+                }
+
+                protected override void ConfigureHowToFindSaga(MessagePropertyMapper<SagaData> mapper)
+                {
+                    mapper.ConfigureMapping<StartSagaMessage>(m => m.Key);
```

Comment:
Correlate with? `Map` and `mapping` are very overloaded words in computer science.